sycl/source/detail/scheduler/commands.cpp

```diff
@@ -2341,12 +2342,22 @@ static pi_result SetKernelParamsAndLaunch(
   }
   if (OutEventImpl != nullptr)
     OutEventImpl->setHostEnqueueTime();
-  pi_result Error = Plugin->call_nocheck<PiApiKind::piEnqueueKernelLaunch>(
-      Queue->getHandleRef(), Kernel, NDRDesc.Dims, &NDRDesc.GlobalOffset[0],
-      &NDRDesc.GlobalSize[0], LocalSize, RawEvents.size(),
-      RawEvents.empty() ? nullptr : &RawEvents[0],
-      OutEventImpl ? &OutEventImpl->getHandleRef() : nullptr);
-  return Error;
+  if (IsCooperative) {
```

Comment:
I'm a little surprised that this branch (and similar branches) ended up using run-time conditions.  One of the benefits of the compile-time properties interface is that `parallel_for` knows at compile-time whether the kernel requires a cooperative launch or not.  The current solution in this PR is turning that compile-time information into run-time information, which seems like it could add some overhead.

I'm happy to defer this sort of thing to later optimizations, but I'm curious to hear whether you think there's a way that we could carry this compile-time information through.  Would it be a very heavy lift, because of the way that we've chosen to represent kernels at the pi and UR level?